test/integ/change_password_integ_test.js

```diff
@@ -380,6 +380,113 @@ describe("integ - change password", () => {
+                        })
+                        .catch((err) => {
+                            reject(err);
+                            stub.restore();
+                        });
+                })
+                .catch((err) => {
+                    reject(err);
+                });
+        });
+    });
+    it("should not change user password if user not found in database", () => {
+        return performAsyncTest((resolve, reject) => {
+            performUserLogin()
+                .then(async (agent) => {
+                    await checkPassword(TEST_USER, "test");
+                    var stub = sinon.stub(databaseOps, "findUser").callsArgWith(1, undefined, []);
+                    agent.post("/change_password?type=json")
+                        .type("form")
+                        .send({
+                            "oldPassword": "test",
+                            "newPassword": "Qwerty123!",
+                            "newPasswordConfirm": "Qwerty123!"
+                        })
+                        .then((res) => {
+                            assert.equal(res.body.errorID, "userNotFound");
+                            checkPassword(TEST_USER, "test")
+                                .then((res) => {
+                                    assert.fail("Password shouldn't change");
+                                })
+                                .catch((err) => {
+                                    assert.isDefined(err);
+                                    resolve();
+                                });
+                            stub.restore();
+                        })
+                        .catch((err) => {
+                            reject(err);
+                            stub.restore();
+                        });
+                })
+                .catch((err) => {
+                    reject(err);
+                });
+        });
+    });
+    it("should change user password if user password change successful but action history cannot write entry due to error", () => {
+        return performAsyncTest((resolve, reject) => {
+            performUserLogin()
+                .then(async (agent) => {
+                    await checkPassword(TEST_USER, "test");
+                    var stub = sinon.stub(actionHistory, "writeActionHistory").callsArgWith(1, new Error("Could not write action history entry"), null);
+                    const newPassword = "Qwerty123!";
+                    agent.post("/change_password?type=json")
+                        .type("form")
+                        .send({
+                            "oldPassword": "test",
+                            "newPassword": newPassword,
+                            "newPasswordConfirm": newPassword
+                        })
+                        .then((res) => {
+                            assert.equal(res.statusCode, 200);
+                            sinon.assert.calledOnce(stub);
```

Comment:
Find a way to assert that the error was passed to callback.

Reply:
Should not be necessary. From the docs about `callsArg`:
> Causes the stub to call the argument at the provided index as a callback function.
and `callsArgWIth`:
> Like callsArg, but with arguments to pass to the callback.
https://sinonjs.org/releases/latest/stubs/